### PR TITLE
feature/api-javascript

### PR DESCRIPTION
**Pull request by william-monroy**

### CODE REVIEW

File: docs/guide/api-javascript.md [hunk 1]
```diff
@@ -32,102 +32,102 @@ const { createServer } = require('vite')
 
 ## `InlineConfig`
 
-The `InlineConfig` interface extends `UserConfig` with additional properties:
+La interfaz `InlineConfig` hereda `UserConfig` con propiedades adicionales:
```
Comment (mrcego) [inline docs/guide/api-javascript.md:35]: Hereda -> extiende

File: docs/guide/api-javascript.md [hunk 2]
```diff
@@ -32,102 +32,102 @@ const { createServer } = require('vite')
...
 
-- `configFile`: specify config file to use. If not set, Vite will try to automatically resolve one from project root. Set to `false` to disable auto resolving.
-- `envFile`: Set to `false` to disable `.env` files.
+- `configFile`: Especifica el archivo de configuración a utilizar. Si no se establece, Vite tratará de resolver automáticamente uno de la raíz del proyecto. Establezca `false` para desactivar la resolución automática.
```
Comment (mrcego) [inline docs/guide/api-javascript.md:37]: Establezca -> establecer a

File: docs/guide/api-javascript.md [hunk 3]
```diff
@@ -32,102 +32,102 @@ const { createServer } = require('vite')
...
+- `envFile`: Establezca `false` para desactivar los archivos `.env`.
```
Comment (mrcego) [inline docs/guide/api-javascript.md:38]: Establezca -> establecer a

File: docs/guide/api-javascript.md [hunk 4]
```diff
@@ -32,102 +32,102 @@ const { createServer } = require('vite')
...
+   * El objeto de configuración de Vite resultante.
    */
   config: ResolvedConfig
   /**
-   * A connect app instance
-   * - Can be used to attach custom middlewares to the dev server.
-   * - Can also be used as the handler function of a custom http server
-   *   or as a middleware in any connect-style Node.js frameworks.
+   * Una instancia de aplicación de conexión
+   * - Puede utilizarse para adjuntar middlewares personalizados al servidor dev.
```
Comment (mrcego) [inline docs/guide/api-javascript.md:50]: al servidor de desarrollo

File: docs/guide/api-javascript.md [hunk 5]
```diff
@@ -32,102 +32,102 @@ const { createServer } = require('vite')
...
+   * Resolver, cargar y transformar programáticamente una URL y obtener el resultado
+   * sin pasar por el pipeline de peticiones http.
    */
   transformRequest(
     url: string,
     options?: TransformOptions
   ): Promise<TransformResult | null>
   /**
-   * Apply Vite built-in HTML transforms and any plugin HTML transforms.
+   * Aplique las transformaciones HTML integradas en Vite y las transformaciones HTML de cualquier plugin.
```
Comment (mrcego) [inline docs/guide/api-javascript.md:89]: Aplique -> Aplica

File: docs/guide/api-javascript.md [hunk 6]
```diff
@@ -32,102 +32,102 @@ const { createServer } = require('vite')
...
    */
   transformIndexHtml(url: string, html: string): Promise<string>
   /**
-   * Load a given URL as an instantiated module for SSR.
+   * Cargar una URL dada como un módulo instanciado para SSR.
```
Comment (mrcego) [inline docs/guide/api-javascript.md:93]: Cargar -> Carga

File: docs/guide/api-javascript.md [hunk 7]
```diff
@@ -32,102 +32,102 @@ const { createServer } = require('vite')
...
-   * Start the server.
+   * Inicia el servidor.
    */
   listen(port?: number, isRestart?: boolean): Promise<ViteDevServer>
   /**
-   * Restart the server.
+   * Reinicia el servidor.
    *
-   * @param forceOptimize - force the optimizer to re-bundle, same as --force cli flag
+   * @param forceOptimize - forzar al optimizador a recomponer el paquete, igual que la bandera --force cli
```
Comment (mrcego) [inline docs/guide/api-javascript.md:110]: Forzar -> Fuerza

File: docs/guide/api-javascript.md [hunk 8]
```diff
@@ -32,102 +32,102 @@ const { createServer } = require('vite')
...
    */
   restart(forceOptimize?: boolean): Promise<void>
   /**
-   * Stop the server.
+   * Detener el servidor.
```
Comment (mrcego) [inline docs/guide/api-javascript.md:114]: Detiene

**Review by mrcego — CHANGES_REQUESTED** [review]

File: docs/guide/api-javascript.md (hunk 1 above)
Comment (william-monroy) [reply docs/guide/api-javascript.md:35]: done

File: docs/guide/api-javascript.md (hunk 2 above)
Comment (william-monroy) [reply docs/guide/api-javascript.md:37]: done

File: docs/guide/api-javascript.md (hunk 3 above)
Comment (william-monroy) [reply docs/guide/api-javascript.md:38]: done

File: docs/guide/api-javascript.md (hunk 4 above)
Comment (william-monroy) [reply docs/guide/api-javascript.md:50]: done

File: docs/guide/api-javascript.md (hunk 5 above)
Comment (william-monroy) [reply docs/guide/api-javascript.md:89]: done

File: docs/guide/api-javascript.md (hunk 6 above)
Comment (william-monroy) [reply docs/guide/api-javascript.md:93]: done

File: docs/guide/api-javascript.md (hunk 7 above)
Comment (william-monroy) [reply docs/guide/api-javascript.md:110]: done

File: docs/guide/api-javascript.md (hunk 8 above)
Comment (william-monroy) [reply docs/guide/api-javascript.md:114]: done

**Comment by william-monroy** [comment]
Requested changes are done. :octocat:

**Pull request merged**